### PR TITLE
Move sorter functionality to pkg zson

### PR DESCRIPTION
**Pull request by mattnibs**
This will be useful in other places. Also added sorter.Pop/Push to
satisfy the demands of heap.Interface.

### CODE REVIEW

File: proc/utils.go [hunk 1]
```diff
@@ -56,6 +56,10 @@ type TestSource struct {
 	idx     int
 }
 
+func NewTestSource(batches []zson.Batch) *TestSource {
```
Comment (henridf) [inline proc/utils.go]: This kind of stuff should ideally be in separate PR (or different commit in this PR would work if we didn't squash commits).

File: proc/utils.go [hunk 2]
```diff
@@ -75,21 +79,30 @@ func (t *TestSource) Parents() []Proc { return nil }
...
 type ProcTest struct {
-	ctx          Context
+	ctx          *Context
 	compiledProc Proc
 	eos          bool
 }
 
-func NewProcTest(code string, resolver *resolver.Table, inRecords []zson.Batch) (*ProcTest, error) {
-	ctx := Context{
+func NewProcTestWithProc(proc Proc, ctx *Context) *ProcTest {
```
Comment (henridf) [inline proc/utils.go]: Since this is a proc unit testing framework, passing a proc object seems like the common case over passing zql source. So I think this should be `NewProcTest` and `NewProcTest` below should become `NewProcTestFromSource` (or some other name). 

But... author has the last word.

**Comment by aswan** [comment]
> This will be useful in other places.

Just curious, what other places?

File: proc/utils.go (hunk 2 above)
Comment (mattnibs) [reply proc/utils.go]: I agree.

File: proc/utils.go (hunk 1 above)
Comment (mattnibs) [reply proc/utils.go]: I was a little lazy here but my `TopProc` pr which is dependent on this, needed this.

Comment (mattnibs) [reply proc/utils.go]: And I figured better to include this then not have a unit test for the other pr.

Comment (nwt) [reply proc/utils.go]: I'm with @henridf because, as far as I can tell, you could write sort_test.go without doing any surgery on this file.

Comment (nwt) [reply proc/utils.go]: But I won't be cranky if you ignore me!

Comment (mattnibs) [reply proc/utils.go]: I'm ignoring you guys now just because I'm trying to gsd the first release of zq, but your concerns are duly noted.

File: pkg/zson/sort.go [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+		dir:     dir,
+		fields:  fields,
+		sorters: make([]sortFn, len(fields)),
+	}
+}
+
+// SortStable performs a stable sort on the provided records.
+func (s *Sorter) SortStable(records []*Record) {
+	s.records = records
+	sort.Stable(s)
```
[the remark this answers is not in the document]
Comment (mattnibs) [reply pkg/zson/sort.go]: Yeah its kind of the weird part of this api.
